src/main/java/com/monitorjbl/xlsx/impl/StreamingSheetReader.java

```diff
@@ -165,8 +183,10 @@ private void handleEvent(XMLEvent event) throws SAXException {
         currentCell.setContents(formattedContents());
       } else if("row".equals(tagLocalName) && currentRow != null) {
         rowCache.add(currentRow);
+        currentRowNum++;
       } else if("c".equals(tagLocalName)) {
         currentRow.getCellMap().put(currentCell.getColumnIndex(), currentCell);
+        currentColNum++;
```

Comment:
The first spreadsheet cannot be represented by just omitting r attribute values. In this case you have to specify r values.
The last spreadsheet can be represented just by omitting r attribute values.
By the way: in my test case I use a spreadsheet where for the first 2 rows no r is specified and for the 3rd row r=7. When you open this spreadsheet in Excel or Libre Office, that the 1st row is displayed at the 1st row. The 2nd one on the 2nd row, and the 3rd row on the 7th row.
So concluding: the implicit rule is when there is no r attribute for a cell or row the following cell/row is used. And the first cell/row without r always start at cell 1 / row 1.
So there is no ambiguity.